pkgs/development/python-modules/pytorch/default.nix

```diff
@@ -122,18 +120,28 @@ in buildPythonPackage rec {
     repo   = "pytorch";
     rev    = "v${version}";
     fetchSubmodules = true;
-    sha256 = "1biyq2p48chakf2xw7hazzqmr5ps1nx475ql8vkmxjg5zaa071cz";
+    sha256 = "07ga9806c5q02hsdybwdw9nc0r1a8dinwjakyg6wlrccclc027lq";
   };
 
-  dontUseCmakeConfigure = true;
-
   preConfigure = lib.optionalString cudaSupport ''
     export TORCH_CUDA_ARCH_LIST="${lib.strings.concatStringsSep ";" final_cudaArchList}"
     export CC=${cudatoolkit.cc}/bin/gcc CXX=${cudatoolkit.cc}/bin/g++
   '' + lib.optionalString (cudaSupport && cudnn != null) ''
     export CUDNN_INCLUDE_DIR=${cudnn}/include
   '';
 
+  # Use pytorch's custom configurations
+  dontUseCmakeConfigure = true;
+
+  BUILD_NAMEDTENSOR = true;
+  BUILD_DOCS = buildDocs;
+  USE_MKLDNN = mklSupport;
```

Comment:
It seems that `mkl`, `mklSupport` and MKLDNN are entangled in this derivation. However, it is possible to build MKLDNN without MKL. MKDNN *can* use an external BLAS implementation, which can be MKL.

So, it seems that there should be a separate `mklDnnSupport` flag, which should probably be set to `true` by default?

Reply:
I'll have to double-check this -- in a former derivation, I recall them being tightly coupled in the pytorch build.

Reply:
I compiled this PR with the following change:

```
diff --git a/pkgs/development/python-modules/pytorch/default.nix b/pkgs/development/python-modules/pytorch/default.nix
index bb98f4ecffc..817141abc10 100644
--- a/pkgs/development/python-modules/pytorch/default.nix
+++ b/pkgs/development/python-modules/pytorch/default.nix
@@ -1,6 +1,7 @@
 { stdenv, fetchurl, fetchgit, buildPythonPackage, python, pythonOlder,
   cudaSupport ? false, cudatoolkit ? null, cudnn ? null, nccl ? null, magma ? null,
   mklSupport ? false, mkl ? null, openblas ? null,
+  mklDnnSupport ? true,
   openMPISupport ? false, openmpi ? null,
   buildBinaries ? false,
   buildDocs ? false,
@@ -135,7 +136,7 @@ in buildPythonPackage rec {
 
   BUILD_NAMEDTENSOR = true;
   BUILD_DOCS = buildDocs;
-  USE_MKLDNN = mklSupport;
+  USE_MKLDNN = mklDnnSupport;
   preBuild = with lib.strings; concatStringsSep "\n" [
     ''export CMAKE_PREFIX_PATH="$CMAKE_PREFIX_PATH:${if mklSupport then mkl else openblas}"''
     "python setup.py build --cmake-only"
```

And it works as expected, without any dependency on Intel MKL.

Reply:
Thanks for the nudge, @danieldk -- I've amended this commit with your suggestion.